Sources/UIKit+Diff.swift

```diff
@@ -88,29 +66,89 @@ public extension UITableView {
         apply(diff: diff.diff, insertRowAnimation: insert, deleteRowAnimation: delete)
         updateVisible(rows: diff.updates, updateVisibleRow)
     }
+
+    public func applyWithoutBatch(diff: Set<DiffStep<Int>>, section: Int,
+                                  insertRowAnimation: UITableViewRowAnimation = .automatic,
+                                  deleteRowAnimation: UITableViewRowAnimation = .automatic) {
+        var insertionIndexPaths: [IndexPath] = []
+        var deletionIndexPaths: [IndexPath] = []
+        for step in diff {
+            switch step {
+            case .insert(let index):
+                insertionIndexPaths.append(IndexPath(row: index, section: section))
+            case .delete(let index):
+                deletionIndexPaths.append(IndexPath(row: index, section: section))
+            case .move(let from, let to):
+                moveRow(at: IndexPath(row: from, section: section),
+                        to: IndexPath(row: to, section: section))
+            }
+        }
+        insertRows(at: insertionIndexPaths, with: insertRowAnimation)
+        deleteRows(at: deletionIndexPaths, with: deleteRowAnimation)
+    }
+
+    public func applyWithoutBatch(diff: Set<DiffStep<IndexPath>>,
+                                  insertRowAnimation: UITableViewRowAnimation = .automatic,
+                                  deleteRowAnimation: UITableViewRowAnimation = .automatic) {
+        var insertionIndexPaths: [IndexPath] = []
+        var deletionIndexPaths: [IndexPath] = []
+        for step in diff {
+            switch step {
+            case .insert(let index):
+                insertionIndexPaths.append(index)
+            case .delete(let index):
+                deletionIndexPaths.append(index)
+            case .move(let from, let to):
+                moveRow(at: from, to: to)
+            }
+        }
+        insertRows(at: insertionIndexPaths, with: insertRowAnimation)
+        deleteRows(at: deletionIndexPaths, with: deleteRowAnimation)
+    }
 }
 
 public extension UICollectionView {
+    public func applyWithoutBatch(diff: Set<DiffStep<Int>>, section: Int) {
```

Comment:
I like having the default of 0 for the general case of `apply`... I see your point here and wonder if it's prudent or foolish to have a consistent API. I think this is fine though.